java/src/main/java/com/google/spsproject/servlets/User.java

```diff
@@ -0,0 +1,96 @@
+package com.google.spsproject.servlets;
+
+import com.google.spsproject.data.UserInfo;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse; 
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.api.core.ApiFuture;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.util.logging.FileHandler;
+import java.util.logging.SimpleFormatter;
+
+@WebServlet("/user")
+public class User extends HttpServlet {
+    private Firestore db;
+    private static final Logger logger = Logger.getLogger(User.class.getName());
+    private FileHandler fh;
+    public User() throws IOException {
+        FirestoreOptions firestoreOptions =
+        FirestoreOptions.getDefaultInstance().toBuilder()
+        .setProjectId("summer20-sps-77")
+        .setCredentials(GoogleCredentials.getApplicationDefault())
+        .build();
+        this.db = firestoreOptions.getService();
+        filehandler = new FileHandler("errorfile.log");
+        logger.addHandler(filehandler);
+        SimpleFormatter formatter = new SimpleFormatter();
+        filehandler.setFormatter(formatter);
+    }
+    public static boolean ValidateString(String str) {
+        if (str != null && !str.trim().isEmpty()) return true;
+        return false;
+    }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String email = request.getParameter("email");
+        DocumentReference docRef = db.collection("users").document(email);
+        ApiFuture<DocumentSnapshot> future = docRef.get();
+        UserInfo UserObject = new UserInfo();
```

Comment:
first character of instance variable should always be small case, otherwise it becomes very difficult to distinguish between a class and an object and consequently a static and non-static method while reading